x-pack/plugins/security/server/plugin.ts

```diff
@@ -48,7 +48,15 @@ export interface LegacyAPI {
  * Describes public Security plugin contract returned at the `setup` stage.
  */
 export interface SecurityPluginSetup {
-  authc: Authentication;
+  authc: Pick<
+    Authentication,
+    | 'isAuthenticated'
```

Comment:
> We can do this in a different PR (or not at all), but this is IMO a rather verbose way of declaring our public authc contract. 

I actually like verbosity since a) it's obvious what exactly plugin consumers will have access to b) we don't accidentally expose new properties/methods (at least not at the type level, like we've done already for methods I'm trying to "deexpose" in this PR).

What does concern you the most in this level of verbosity?

> It might be clearer if the Authentication service included an internal property (or similar), which could contain the properties we don't intend to expose to consumers.

But since we still shouldn't expose `internal` we'll have to "omit" it by specifically picking non-internal properties before exposing anyway. Also technically `AuthenticationService` doesn't know what from what it returns is internal and what isn't, it's up to the main routine (Plugin) to decide. 

Having said that we can create a dedicated interface for public part of `Authentication` contract (i.e. dedicated interface vs `Pick<Authentication, ...>`) so that `Authentication` isn't visible to consumers at all, would it help?

Reply:
> What does concern you the most in this level of verbosity?
>
> Having said that we can create a dedicated interface for public part of Authentication contract (i.e. dedicated interface vs Pick<Authentication, ...>) so that Authentication isn't visible to consumers at all, would it help?

yeah I think I would like that better. Again, nothing we have to do in this PR if you'd rather defer. This all comes down to personal taste, I have no real concerns with the approach

Reply:
Grr, wanted to introduce a new interface and rename the old one in this PR, but it's going to affect lots of unrelated places. I'd rather make it in a dedicated PR if you don't mind. 

Btw, I've just checked how Core deals with this: they have a pair of interfaces that look like this:
```typescript
ElasticsearchServiceSetup  {...}

/** @internal */
interface InternalElasticsearchServiceSetup extends ElasticsearchServiceSetup  {...}
```

Reply:
Yep, dedicated PR would be perfect! No rush on that either, I know we have more pressing matters